qhana_plugin_registry/db/models/templates.py

```diff
@@ -231,6 +267,34 @@ class TemplateTab(IdMixin, ExistsMixin, NameDescriptionMixin):
     def plugin_filter(self) -> dict:
         return json.loads(self.filter_string)
 
+    @classmethod
+    def get_by_name(cls, name: str) -> "Optional[TemplateTab]":
+        q = select(cls).filter(cls.name == name)
+        found_tab: Optional[TemplateTab] = DB.session.execute(q).scalar_one_or_none()
+        return found_tab
+
+    @classmethod
+    def get_or_create(cls, tab: dict) -> "TemplateTab":
+        found_tab: Optional[TemplateTab] = cls.get_by_name(tab["name"])
+        if found_tab is None:
+            found_tab = cls(
+                name=tab["name"],
+                description=tab["description"],
+                sort_key=tab["sort_key"],
+                location=tab["location"],
+                filter_string=json.dumps(tab["filter"]),
+                template=tab["template"],
+            )
+            DB.session.add(found_tab)
+            DB.session.commit()
+        return found_tab
+
+    @classmethod
+    def get_or_create_all(cls, tabs: Sequence[dict]) -> "List[TemplateTab]":
+        if not tabs:
+            return []
+        return [cls.get_or_create(tab) for tab in tabs]
```

Comment:
If you want to have an autocommit for this function, then it should only happen once after all tabs were created (and not after each individual tab).